keras_nlp/models/bert/bert_classifier.py

```diff
@@ -190,6 +190,14 @@ def __init__(
         self.num_classes = num_classes
         self.dropout = dropout
 
+        # Default compilation
+        self.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(5e-5),
```

Comment:
Overall this LGTM! This will clearly be an improvement over the default optimizer in core keras, and from `optimizer="adam"`, so the delta is good. Two questions we may want to consider...

Should we try to use `AdamW`? My thinking is probably no right now, because `AdamW` only works well in tf 2.10, and colab tf is still 2.9. AdamW does seem to be the de-facto choice for BERT, RoBERTa, and most of our other models in the "upstream version."

Should we do any sort of decay in the learning rate that is not tied to dataset cardinality? T5 used an "inverse square root" learning rate when they we searching for a "train step independent" learning rate. Overall though, I think keeping it simple and getting feedback sgtm.

Reply:
The issue with LR schedule is that you need to know the number of steps ([example](https://github.com/keras-team/keras-nlp/blob/master/examples/glue_benchmark/scores.md#hyperparameter-settings)). How would we make this general?

Reply:
Yeah for sure, that's why I was mentioning decay **not** tied to train steps. T5 uses `1 / sqrt(max(n, k))` where `n` is the current step, and `k` is the warmup steps. They mentioned in the paper that this is worse than the triangular learning rate, but they chose it because it was general purpose. You'll notice that there is no part of that the includes the total number of train steps!

We probably have no way to include a warm-up here (because who know how many warmup steps is correct), but you could imaging a exponentially decaying schedule where the learning rate constantly falls but never zeros. That **could** be something that would lead to better performance on datasets with, say, 100k+ examples. I believe @chenmoneygithub was mentioning that for MNLI which is roughly 400k example, having some decay was important!

This is definitely not something something we need to solve on this PR, but it is something we should consider here. Is a flat forever learning rate here "just bad" in many cases, and if so, can we do better without knowing `train_steps`?

Reply:
Great point! Let's file an issue. FWIW I've not had any issues fine-tuning with this setup for BertClassifier.

Reply:
The place I know a lr cooldown was useful was mnli. So if you haven't fine tuned anything larger than imdb, that may be fully expected.

But sounds good! Will file an issue.